### PR TITLE
ASVS metadata

### PR DESCRIPTION
**Pull request by inkz**
commits from https://github.com/returntocorp/semgrep-rules/pull/1175
something stuck on original pull request

### CODE REVIEW

**Comment by inkz** [comment]
original pull request https://github.com/returntocorp/semgrep-rules/pull/1175 by @hazanasec 
I merged it with the current `develop` and added `pre-commit` linting

**Comment by colleend** [comment]
huh, strange that the upload of the artifact failed 🤔

File: generic/secrets/security/detected-npm-registry-auth-token.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+rules:
+- id: detected-npm-registry-auth-token
+  pattern-regex: =(\w{8})-(\w{4})-(\w{4})-(\w{4})-(\w{12})$
```
Comment (clintgibler) [inline generic/secrets/security/detected-npm-registry-auth-token.yaml:3]: I feel like we can make this more precise, like scoping it to an npmrc file or something, and maybe including `_authToken=` in the pattern?

**Review by clintgibler — APPROVED** [review]
LGTM, let's merge it in if @inkz or @minusworld agree 👍

**Comment by inkz** [comment]
resubmitted: https://github.com/returntocorp/semgrep-rules/pull/1207

**Pull request closed without merging**